src/Guard.php

```diff
@@ -159,6 +159,9 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
         // Generate new CSRF token if persistentTokenMode is false, or if a valid keyPair has not yet been stored
         if (!$this->persistentTokenMode || !$this->loadLastKeyPair()) {
             $request = $this->generateNewToken($request);
+        } elseif ($this->persistentTokenMode) {
```

Comment:
Indeed it's not really easy to read, but I don't see a prettier way to put it right. Any suggestions?